staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go

```diff
@@ -107,6 +107,15 @@ func MonitorRequest(request *http.Request, verb, resource, subresource, scope, c
 		}
 	}
 
+	// make get pod log with follow option api request reported as watch
+	if verb == "GET" && resource == "pods" && subresource == "log" {
```

Comment:
This problem is not the logic, the problem is that we cannot add such a hack in a generic library. The next one with a `follow` URL param will be surprised that the reported verb is wrong. It's just the wrong layer of the system

The cited `?watch` syntax is `k8s.io/apiserver` specific, not Kubernetes specific in the sense of the container orchestrator.

We could possibly add the request URL param keys to the metrics in a generic way. But we cannot add special handling for pods/log here.